packages/shylu/basker/src/basker_matrix_decl.hpp

```diff
@@ -46,13 +46,19 @@ namespace BaskerNS
     //init_matrix (want to change these to malloc_matrix)
     BASKER_INLINE
     void init_matrix(string _label, Int _m, Int _n, Int _nnz);
+
     BASKER_INLINE
     void init_matrix(string _label, Int _m, Int _n, Int _nnz,
                     Int *_col_ptr, Int *_row_idx, Entry *_val);
     BASKER_INLINE
     void init_matrix(string _label, Int _sr, Int _m, 
                     Int _sc, Int _n, Int _nnz);
 
+    template <typename T2>
```

Comment:
Make private; want to make Amesos2 a 'friend' to deal with discrepancy of integral types returned by Tpetra (i.e. row ptr type different from col indices)...